From 06e32f1d644ab259064a89368cded79c1af5fd16 Mon Sep 17 00:00:00 2001
From: Bin Han <bin.b.han@intel.com>
Date: Mon, 23 Mar 2015 10:42:03 +0800
Subject: [PATCH 323/323] atomisp2: turn off enable_yuv_ds if YUV DS is not
 needed

Previously, enable_yuv_ds is always turned on for preview
pipe, this is not correct if we do not actually neeed YUV
downscaling. This patch checks if the vf_pp_in_res is same
to the output resolutions, if it is true, turn off the
enable_yuv_ds and set vf_pp_is_res to 0.

Change-Id: I2d85702df72136ac1bc1d6fe3fb435998fba46b5
Tracked-On: #H1603950923
Signed-off-by: Bin Han <bin.b.han@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/47405
Reviewed-by: Tian Shu Qiu <tian.shu.qiu@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Approver: Tian Shu Qiu <tian.shu.qiu@intel.com>
Integrator: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c  | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index 944d1ef..9ac7d2a 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -2703,8 +2703,6 @@ static void __configure_preview_pp_input(struct atomisp_sub_device *asd,
 	pipe_configs->mode = __pipe_id_to_pipe_mode(asd, pipe_id);
 	stream_env->update_pipe[pipe_id] = true;
 
-	pipe_extra_configs->enable_yuv_ds = true;
-
 	out_width = pipe_configs->output_info[0].res.width;
 	out_height = pipe_configs->output_info[0].res.height;
 
@@ -2781,6 +2779,15 @@ static void __configure_preview_pp_input(struct atomisp_sub_device *asd,
 		}
 	}
 
+	if (vf_pp_in_res->width == out_width &&
+		vf_pp_in_res->height == out_height) {
+		pipe_extra_configs->enable_yuv_ds = false;
+		vf_pp_in_res->width = 0;
+		vf_pp_in_res->height = 0;
+	} else {
+		pipe_extra_configs->enable_yuv_ds = true;
+	}
+
 	dev_dbg(isp->dev, "configuring pipe[%d]preview pp input w=%d.h=%d.\n",
 		pipe_id, width, height);
 }
-- 
1.9.1

